nixos/modules/services/networking/seafile.nix

```diff
@@ -269,30 +385,76 @@ in
           preStart = ''
             mkdir -p ${seahubDir}/media
             # Link all media except avatars
-            for m in `find ${pkgs.seahub}/media/ -maxdepth 1 -not -name "avatars"`; do
+            for m in `find ${seahub}/media/ -maxdepth 1 -not -name "avatars"`; do
               ln -sf $m ${seahubDir}/media/
             done
             if [ ! -e "${seafRoot}/.seahubSecret" ]; then
-                ${pkgs.seahub.python}/bin/python ${pkgs.seahub}/tools/secret_key_generator.py > ${seafRoot}/.seahubSecret
+                ${python}/bin/python ${seahub}/tools/secret_key_generator.py > ${seafRoot}/.seahubSecret
                 chmod 400 ${seafRoot}/.seahubSecret
             fi
             if [ ! -f "${seafRoot}/seahub-setup" ]; then
                 # avatars directory should be writable
-                install -D -t ${seahubDir}/media/avatars/ ${pkgs.seahub}/media/avatars/default.png
-                install -D -t ${seahubDir}/media/avatars/groups ${pkgs.seahub}/media/avatars/groups/default.png
+                install -D -t ${seahubDir}/media/avatars/ ${seahub}/media/avatars/default.png
+                install -D -t ${seahubDir}/media/avatars/groups ${seahub}/media/avatars/groups/default.png
                 # init database
-                ${pkgs.seahub}/manage.py migrate
+                ${seahub}/manage.py migrate
                 # create admin account
-                ${pkgs.expect}/bin/expect -c 'spawn ${pkgs.seahub}/manage.py createsuperuser --email=${cfg.adminEmail}; expect "Password: "; send "${cfg.initialAdminPassword}\r"; expect "Password (again): "; send "${cfg.initialAdminPassword}\r"; expect "Superuser created successfully."'
-                echo "${pkgs.seahub.version}-sqlite" > "${seafRoot}/seahub-setup"
+                ${pkgs.expect}/bin/expect -c 'spawn ${seahub}/manage.py createsuperuser --email=${cfg.adminEmail}; expect "Password: "; send "${cfg.initialAdminPassword}\r"; expect "Password (again): "; send "${cfg.initialAdminPassword}\r"; expect "Superuser created successfully."'
+                echo "${seahub.version}-mysql" > "${seafRoot}/seahub-setup"
             fi
             if [ $(cat "${seafRoot}/seahub-setup" | cut -d"-" -f1) != "${pkgs.seahub.version}" ]; then
-                # update database
-                ${pkgs.seahub}/manage.py migrate
-                echo "${pkgs.seahub.version}-sqlite" > "${seafRoot}/seahub-setup"
+                # run django migrations
+                ${seahub}/manage.py migrate
+                echo "${seahub.version}-mysql" > "${seafRoot}/seahub-setup"
             fi
           '';
         };
+
+        seaf-gc = {
```

Comment:
it'd be nice if this was closer to the [`nix.gc` module](https://github.com/NixOS/nixpkgs/blob/9fb66c698910d30a9e41b53b6ba79a55c9da1a9c/nixos/modules/services/misc/nix-gc.nix) with it's timer

Reply:
well, we don't want them to run at the same time ideally.

Reply:
I reformatted the gc options into its own attrset, and added the options that nix.gc presented.